dash/include/dash/Atomic.h

```diff
@@ -202,19 +204,288 @@ class Atomic
    */
   ValueType fetch_and_sub(
     /// Value to be subtracted from global atomic variable.
-    ValueType val)
+    ValueType value)
+  {
+    return fetch_and_op(dash::plus<ValueType>(), -value);
+  }
+  
+  /**
+   * Atomically fetches and replaces the value.
+   * 
+   * \return  The value of the referenced shared variable before the
+   *          operation.
+   */
+  ValueType exchange(
+    /// Value to be set
+    ValueType value)
   {
-    return fetch_and_op(dash::plus<ValueType>(), -val);
+    return fetch_and_op(dash::second<ValueType>(), value);
+  }
+  
+  /**
+   * Atomically compares the value with the value of expected and if those are
+   * bitwise-equal, replaces the former with desired.
+   * 
+   * \note not implemented yet
+   * 
+   * \return  True if value is exchanged
+   */
+  bool compare_exchange(
+    /// value expected to be found in the atomic object
+    ValueType expected,
+    /// value to store in the atomic object if it is as expected
+    ValueType desired)
+  {
+    // TODO;
```

Comment:
I guess this requires something like `dart_compare_and_swap()`, will add an implementation of it today.